feat: added more subtitle customization options

lancechant opened the pull request:
<!--
  Pull Request Template for Streamyfin
  ====================================
  Use this template to help reviewers understand the purpose of your PR
  and to ensure all necessary checks are completed before merging.
-->

# 📦 Pull Request

## 🔖 Summary
<!--
A concise description of the changes introduced by this PR.
Example:
“Add real-time currency conversion widget to dashboard.”
-->
Subtitle customizations

## 🏷️ Ticket / Issue
<!--
Link to the related ticket, issue or user story.
You can also indicate if this PR supersedes a previous one.
Example:
- Closes #123
- Fixes STREAMYFIN-456
- Resolves #789
- Supersedes #120
- Related: #130
-->
Closes #1060 

## 🛠️ What’s Changed
<!-- Use a Conventional Commit in the PR title, e.g., `feat(auth): add MFA`. 
If this PR introduces a breaking change, include a `BREAKING CHANGE:` block in the description.
Spec: https://www.conventionalcommits.org/ -->

- Type: feat
Subtitles can now be customized with the following extra options in settings:
- Colour
- background opacity/colour
- outline opacity/colour
- boldness

## 📋 Details
<!--
Provide more context or background. Explain any non-obvious decisions.
Include screenshots or GIFs for UI changes if applicable.
-->

### ⚠️ Breaking Changes
<!-- List any breaking API/contract changes and migration guidance. If none, write “None”. -->

### 🔐 Security & Privacy Impact
<!-- Data touched, new permissions/scopes, PII, secrets, threat considerations. If none, write “None”. -->

### ⚡ Performance Impact
<!-- Hot paths, memory/CPU/latency implications, benchmarks if available. -->

### 🖼️ Screenshots / GIFs (if UI)
<!-- Before/After, dark mode, responsive states. -->
<img width="1080" height="2183" alt="Streamyfin-settings" src="https://github.com/user-attachments/assets/7423aae0-7d37-4235-8cb2-b6b0d2465387" />


## ✅ Checklist
<!--
Review and check off items as you complete them.
-->
- [ ] I’ve read the [contribution guidelines](CONTRIBUTING.md)
- [ ] Code follows project style and passes lint/format (`npm|pnpm|yarn|bun` scripts)
- [ ] Type checks pass (tsc/biome/etc.)
- [ ] Docs updated (README/ADR/usage/API)
- [ ] No secrets/credentials included; env vars documented
- [ ] Release notes/CHANGELOG entry added (if applicable)
- [ ] Verified locally that changes behave as expected

## 🔍 Testing Instructions
<!--
Describe how reviewers can test your changes.
Example:
1. `git fetch origin pull/<PR_ID>/head:branchname && git checkout branchname`
2. Install deps: `npm|pnpm|yarn|bun install`
3. Start service/app: `npm|pnpm|yarn|bun run [target]` (e.g., `npm run ios` or `bun run android:tv`)
4. Run tests: `npm|pnpm|yarn|bun test`
5. Verification steps:
   - [ ] Expected UI/endpoint behavior
   - [ ] Logs show no errors
   - [ ] Edge cases covered (list)
-->

## ⚙️ Deployment Notes
<!--
Describe any deployment considerations such as config, environment vars, or native builds.
-->

## 📝 Additional Notes
<!--
Any other information or references related to this PR.
-->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added comprehensive subtitle styling controls: text color, background color, outline color/thickness, background/outline opacity, bold toggle, and subtitle size scaling; applied during playback.

- Settings
  - Settings panel now includes controls with live previews and a new hint explaining subtitle behavior and appearance.

- Style
  - Updated subtitle labels and color/thickness names; standardized “OnlyForced” label.

- Chores
  - Removed a debug log from storage settings.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

lancechant committed:
feat: added more subtitle customization options

Subtitles can now be customized with the following extra options:
- Colour
- background opacity/colour
- outline opacity/colour
- boldness

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
chore: moved constant values to a file

Reduced duplication and removed constants to a new file

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

Simon-Eklundh reviewed (commented):
Lgtm, will approve when it's up to date

lancechant committed:
Merge branch 'develop' into feature/subtitle-customizations

lancechant commented:
Hey @Simon-Eklundh it's up to date with develop if that's what you mean?

lancechant committed:
Merge branch 'develop' into feature/subtitle-customizations

lancechant committed:
Update direct-player.tsx

Added text scaling

lancechant committed:
Update direct-player.tsx

Fixed syntax

lancechant committed:
Merge remote-tracking branch 'origin/develop' into feature/subtitle-customizations

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

lancechant committed:
chore: rework logic

Changed logic to use the settings component to store the changed values
rather than referencing storage directly
Deleted an unused file

Signed-off-by: Lance Chant <13349722+lancechant@users.noreply.github.com>

fredrikburmester committed:
chore: remove log

fredrikburmester committed:
Merge branch 'develop' into feature/subtitle-customizations
